torch/cuda/__init__.py

```diff
@@ -1137,6 +1137,46 @@ def _dtype(self):
 torch._storage_classes.add(ComplexDoubleStorage)
 torch._storage_classes.add(ComplexFloatStorage)
 
+
+class _WrappedTritonKernel(object):
+    """ Just a simple wrapper to store some metadata for testing purposes.
+    """
+
+    def __init__(self, kernel):
+        self.kernel = kernel
+        self.kernel_invoked = False
+
+    def __call__(self, *args, **kwargs):
+        res = self.kernel(*args, **kwargs)
+        self.kernel_invoked = True
+        return res
+
+
+def _register_triton_kernels():
+    import sys
+    import importlib
+
+    if sys.executable == "torch_deploy":
```

Comment:
One of the last failing tests is related to torch.deploy. The linter suggest to use `if torch._running_with_deploy():` here instead of `if sys.executable == "torch_deploy":`. @nikitaved can you try fixing the lint issue and making this change? Maybe that'll fix the CI errors.

Reply:
Yes that was the logical merge conflict that we were expecting with my PR. `lintrunner -a` should fix it for you :)

Reply:
btw this is the cause for the deploy test as well: `torch.cuda.DeferredCudaCallError: CUDA call failed lazily at initialization with error: module 'torch' has no attribute '_TritonLibrary'`

Reply:
`_running_with_deploy` saved the day :)